contribs/gmf/src/permalink/Permalink.js

```diff
@@ -1543,4 +1494,63 @@ gmf.Permalink.prototype.cleanParams = function(groups) {
 };
 
 
-gmf.module.service('gmfPermalink', gmf.Permalink);
+gmf.permalink.Permalink.module = angular.module('gmfPermalink', [
+  ngeo.statemanager.module.name,
+  gmf.theme.Themes.module.name,
+  gmf.theme.Manager.module.name,
+  ngeo.draw.features.name,
+  ngeo.misc.EventHelper.module.name,
+  ngeo.statemanager.module.name,
+  ngeo.layertree.Controller.module.name,
+]);
+
+gmf.module.requires.push(gmf.permalink.Permalink.module.name);
+
+gmf.permalink.Permalink.module.service('gmfPermalink', gmf.permalink.Permalink);
+
+
+/**
+ * @enum {string}
+ */
+gmf.permalink.Permalink.OpenLayersLayerProperties = {
+  OPACITY: 'opacity'
+};
+
+/**
+ * @enum {string}
+ */
+gmf.permalink.Permalink.ParamPrefix = {
+  DIMENSIONS: 'dim_',
+  TREE_ENABLE: 'tree_enable_',
+  TREE_GROUP_LAYERS: 'tree_group_layers_',
+  TREE_GROUP_OPACITY: 'tree_group_opacity_',
+  TREE_OPACITY: 'tree_opacity_',
+  WFS: 'wfs_'
+};
+
+
+/**
+ * External data source separators
+ * @enum {string}
+ */
+gmf.permalink.Permalink.ExtDSSeparator = {
+  LIST: ',',
+  NAMES: ';'
+};
+
+
+gmf.permalink.Permalink.module.value('gmfPermalinkOptions',
+  /** @type {gmfx.PermalinkOptions} */ ({}));
+
+
+/** Configure the ngeo state manager */
+(function() {
+  const regexp = [];
+  for (const key1 in gmf.permalink.Permalink.ParamPrefix) {
+    regexp.push(new RegExp(`${gmf.permalink.Permalink.ParamPrefix[key1]}.*`));
+  }
+  for (const key2 in gmf.PermalinkParam) {
+    regexp.push(new RegExp(gmf.permalink.Permalink.ParamPrefix[key2]));
+  }
+  ngeo.module.value('ngeoUsedKeyRegexp', regexp);
```

Comment:
Strange to use ngeo instead of gmf here. But go ahead, that's not for this PR.

Reply:
Yes, I left it mostly untouched, just moved it to the bottom.